shared-bindings/socketpool/SocketPool.c

```diff
@@ -86,21 +86,25 @@ MP_DEFINE_EXCEPTION(gaierror, OSError)
 //|
 //|     IP_MULTICAST_TTL: int
 //|
-//|     def socket(self, family: int = AF_INET, type: int = SOCK_STREAM) -> socketpool.Socket:
+//|     def socket(
+//|         self, family: int = AF_INET, type: int = SOCK_STREAM, proto: int = 0
+//|     ) -> socketpool.Socket:
 //|         """Create a new socket
 //|
 //|         :param ~int family: AF_INET or AF_INET6
 //|         :param ~int type: SOCK_STREAM, SOCK_DGRAM or SOCK_RAW
+//|         :param ~int proto: IPPROTO_TCP or IPPROTO_IP. Only works with SOCK_RAW 
```

Comment:
I manually passed the value of proto = 1 for ICMP and it works to allow me to create a manual ICMP packet with a raw socket (I've tested it).  Unfortunately, it doesn't look like IPPROTO_ICMP = 1 is defined anywhere, in the existing code base.